mzLib/Proteomics/RetentionTimePrediction/ChronologerModel/ChronologerEstimator.cs

```diff
@@ -0,0 +1,344 @@
+
+                bool nTerminalMod = fullSequence[0] == '['; // true if the first loop is a mod
+
+                foreach (var subString in fullSeq)
+                {
+                    //if mod, enter
+                    if (nTerminalMod)
+                    {
+                        if (subString.Contains("Acetyl"))
+                            tensor[0][0] = 39;
+                        else
+                        {
+                            tensor[0][0] = 38;
+                        }
+                        nTerminalMod = false; //next iteration is not a mod
+                        continue;
+                    }
+
+                    if (!subString.Contains(" "))
+                    {
+                        //without mods
+                        for (int i = 0; i < subString.Length; i++)
+                        {
+                            tensor[0][tensorCounter] = ChronologerDictionary[(subString[i], "")];
+                            tensorCounter++;
+                        }
+
+                        //save target aa for next loop
+                        modID = subString[subString.Length - 1];
+                    }
+
+                    nTerminalMod = true;
+                }
+
+                tensor[0][tensorCounter] = 44; //C-terminus
+
+                return tensor;
+            }
+
+            return torch.zeros(1, 52, torch.ScalarType.Int64);
+
+        }
+
+        private static readonly Dictionary<(char, string), int> ChronologerDictionary = new()
+            {
+                { ('A', ""), 1 }, //'Alanine
+                { ('C', ""), 2 }, //'Cysteine
+                { ('D', ""), 3 }, //'Aspartate
+                { ('E', ""), 4 }, //'Glutamate
+                { ('F', ""), 5 }, //'Phenylalanine
+                { ('G', ""), 6 }, //'Glycine
+                { ('H', ""), 7 }, //'Histidine
+                { ('I', ""), 8 }, //'Isoleucine
+                { ('K', ""), 9 }, //'Lysine
+                { ('L', ""), 10 }, //'Leucine
+                { ('M', ""), 11 }, //'Methionine
+                { ('N', ""), 12 }, //'Asparagine
+                { ('P', ""), 13 }, //'Proline
+                { ('Q', ""), 14 }, //'Glutamine
+                { ('R', ""), 15 }, //'Argenine
+                { ('S', ""), 16 }, //'Serine
+                { ('T', ""), 17 }, //'Threonine
+                { ('V', ""), 18 }, //'Valine
+                { ('W', ""), 19 }, //'Tryptophane
+                { ('Y', ""), 20 }, //'Tyrosine
+                { ('C', "Carbamidomethyl on C"), 21 }, //'Carbamidomethyl
+                { ('M', "Oxidation on M"), 22 }, //'Oxidized
+                { ('E', "Glu to PyroGlu"), 24 }, //'Pyro-glutamate
+                { ('S', "Phosphorylation on S"), 25 }, //'Phosphoserine
+                { ('T', "Phosphorylation on T"), 26 }, //'Phosphothreonine
+                { ('Y', "Phosphorylation on Y"), 27 }, //'Phosphotyrosine
+                { ('K', "Accetylation on K"), 28 }, //acetylation
```

Comment:
What about alternative spellings? Akin to the issue from Zhuoxin's two phosph on S bars in her presentation. Unimod may say: Phospho on S where we may say Phosphorylation on S